### ISSUE TITLE
Is learning rate decay for Adam implemented properly?

### ISSUE THREAD

**Issue by Threynaud** [marian-nmt/marian-dev#419]
This paper: [Decoupled Weight Decay Regularization](https://arxiv.org/abs/1711.05101) states that the implementation of weight decay for Adam has been wrong in a lot of deep learning frameworks and libraries.

Has the fix that they suggest in the paper been implemented in Marian already? It could significantly increase the performance of certain architectures.

**Comment by emjotde** [comment]
Hi, we have it for Adam. 
With`--optimizer adam --optimizer-params 0.9 0.999 1e-5 0.5` the 0,5 weight would enable weight decay in AdamW. Otherwise the weight is 0 being standard Adam.

**Comment by emjotde** [comment]
That being said, it doesn't do much for me, seems to converge a bit faster in the beginning.

**Comment by Threynaud** [comment]
Perfect, thanks a lot!
